src/handler/mod.rs

```diff
@@ -853,7 +967,7 @@ impl Handler {
                 }
                 let id = request.id.clone();
                 trace!("Sending next awaiting message. Node: {}", contact);
-                if let Err(request_error) = self.send_request(contact, request).await {
+                if let Err(request_error) = self.send_request(contact, request, false).await {
```

Comment:
this request might have been queued with a hole punch = true. I think it either does not need to be queued or the values needs to be reused?